Proof reading - competencies.md (#165)
CaptainSifff/paper_teaching-learning-RSE: competencies.md

Moreover, we mention, that several subsections of NFDI engage in domain specific
educational programs.

#### SURESOFT

SURESOFT [@SURESOFTLink] is a DFG funded project at TU Braunschweig and
FAU Erlangen-Nürnberg fostering the sustainability of research software
by helping researchers adopt practices and tools from the software
engineering community [@SURESOFT2022]. Because tools are not enough on their own but require skill-full execution, the project implements a twofold approach that combines tools and

> infrastructure with education in the form of workshops and hands-on training. While training of The Carpentries [@Carpentries] is more targeted towards novices, SURESOFT workshops are focused on more advanced software engineering topics like Software Design Principles, Design Patterns, Refactoring, Continuous Integration and Test Driven Development. 

Reviewer: Suggesting a minor edit:
```suggestion
infrastructure with education in the form of workshops and hands-on training. While the training material provided by The Carpentries [@Carpentries] is more targeted towards novices, SURESOFT workshops are focused on more advanced software engineering topics like Software Design Principles, Design Patterns, Refactoring, Continuous Integration and Test Driven Development. 
```

A lot of RSE work involves the manipulation or creation of data processing devices.
We highlight that professional conduct requires these creations to be reliable and to maintain data integrity.

> In particular, handling personal data may have far reaching implications for society.

Reviewer: I might be inclined to make a minor edit here as follows:
```suggestion
In particular, the way that personal data is handled can have far reaching implications for society.
```

Reviewer 2: I agree with Jeremy.

Reviewer 3: Well then the popular vote wins. It was just gut feeling anyways :)

RSEs often operate at the cutting edge of technological development
and therefore might have to deal with technologies of which the dangers and drawbacks are still poorly understood.
A current example is the rush for the application of Large Language Models (LLMs),

> where RSEs working in these fields should stay up-to-date and be able to help researchers assess topics

Reviewer: I'd probably be inclined to leave this as it was but I guess either way is OK! 🙂 
```suggestion
where RSEs working in these fields should stay up-to-date and be able to help researchers to assess topics
```

### Adapting to the software life-cycle (SWLC)

The traditional software life-cycle defines the stages that form the process of building a piece of software.
This is then followed by testing that things work as expected and that they continue to do so into the future.

> This generally involves a creative process where requirements are gathered and analysed followed by a formulation of a plan to fulfil them that is finally implemented.

Reviewer: ```suggestion
This generally involves a creative process where requirements are gathered and analysed, followed by a formulation of a plan to fulfil them that is finally implemented.
```

Reviewer 2: I do not understand why we switch the order.

Reviewer: From our discussion in the meeting - how about we extend this as follows:
```suggestion
This generally involves a creative process where requirements are gathered and analysed followed by a formulation of a plan to fulfil them that is finally implemented. In modern "agile" software development, this is generally undertaken as an iterative process that supports more effective and reliable software outputs since it can react to and take into account feedback, changes in specification or technical challenges more efficiently.
```

Reviewer 2: I tried my hand at a formulation.

RSEs often need to interact with and facilitate communication among colleagues, clients and contractors
with a very broad spectrum of background-knowledge, specialisation, expectations and experience.
Team skills are therefore crucially important.
Team skills are also mentioned in common guides for software engineering such as the Software Engineering Body of Knowledge [@swebok_2024].
However, the interpersonal and organisational skills and the capacity for adaption required to work in a research setting
warrants a much stronger emphasis on this field of competence.

### Working in a team (TEAM)

Working in a team is all about communication and teamwork.
For example, RSEs need to be able to explain particular implementation choices made and may even need to defend them.
Within a team of RSEs code reviews improve knowledge transfer and increase team cohesion.
Code review within a multi-disciplinary team fosters mutual understanding and
can be a knowledge transfer between the RSE and domain scientists.
The team might change on a project-to-project basis and might be comprised of colleagues with very different backgrounds

> including, next to software engineers, for example, IT staff, domain scientists and technicians.

Reviewer: Not sure about this one - it's not so clear to read but I'm not sure if my suggested modification is much better!
```suggestion
including, for example, IT staff, domain scientists and technicians working alongside software engineeers.
```

Now that we have the different competencies, we can explore various dimensions of these competencies,
depending on their circumstances. A strong beneficiary of specialised RSEs can also be newly formed RSE centres at research institutions.

## Career Level

At different career levels, differing skills are required. We have set this up according to the following separation often applied within a single project:

> - Junior RSE: These are persons who have just started, but generally speaking they should have the skills to contribute to software projects.

Reviewer: Suggesting a slight rewording:
```suggestion
- Junior RSE: These are people who have recently started out on their RSE career journey, but generally speaking they should still have the skills to actively contribute to software projects.
```

#### Bachelor's Level

Students at undergraduate level mostly consume science/knowledge.
During their studies, they should also learn about the existence of digital tools and structures.
Undergrad students should be aware that RSEs exist and that software has different quality aspects (DOCBB).
They should be aware of domain specific tools (LIBS, SRU) and where to find them (SWREPOS, DOMREP).

> At this level, it is sufficient to consider software as black boxes (USERS) although some training in data presentation would be very helpful and a good way to find out about programming (MOD, NEW).

Reviewer: ```suggestion
At this level it is sufficient to consider software as black boxes (USERS) although some training in data presentation would be very helpful and a good way to find out about programming (MOD, NEW).
```

What we have defined above are intended to be base skills that an RSE irrespective of domain, place, and time should know about.
But not all RSEs are created equal, they specialise in different areas,
some of which we want to present below. Many of the specialisations may overlap,
so the same RSE might for example work on data management and Open Science.
We categorise them into those, that can be viewed as a specialisation within RSE-specific topics,
while other RSEs might expand their skill set and profession to areas, that are not typical for an RSE.

### Specialisations within the core RSE competencies

#### OpenScience RSE
Open Science and FAIRness of Data and Software are increasingly important topics in research,
as exemplified by the demand of an increasing amount of research funding agencies requiring openness.

> Hence an Open Science RSE is required to have a deepened knowledge in (RC) and how to distribute software publicly (SRU, SP).

Reviewer: "deepened" -> "deeper"?

Reviewer 2: Basically it'S your call here, to determine what is the better idiomatic english...

#### Legal-RSE
All RSEs are a go-to person for questions about licensing, mixing and matching of software.
But with the rising requirements from legislation,

> we foresee the need for RSEs that still have a background in RSE but extend it with a knowledge of legal processes, that cover corner cases and go beyond applying Best Practice guides.

Reviewer: ```suggestion
we foresee the need for RSEs that still have a background in RSE but extend it with a knowledge of legal processes that cover corner cases and go beyond applying Best Practice guides.
```

If they enjoy this role and would like to take it further they need to know that there is an RSE career path as well as that specialised training materials exist.
Beyond that, for an RSE it is important to be a part of a network with other RSEs, irrespective of the career level.
A perfect first step for forming this network is topic-related conferences, workshops or meet-ups.
Beyond that, there is the broader RSE community organised at the local and regional level with chapters or local/regional communities, at the national level with societies and the international RSE society.
Each of them offers possibilities for connecting within or beyond an individual institution and is a great way to find like-minded people to grow a wider network and thereby facilitate the sharing of information on interesting events or help each other out.
This available layered network can greatly benefit the RSE in finding help with issues outside of their own comfort zone
and provides a welcoming, social safety net providing a home for the RSE. Since we feel providing aspiring RSEs with this net
is of utmost importance we envision compulsory events introducing that to young RSEs.

> Qualification badges are another venue for RSEs to find people with the same technical interests.

Reviewer: ```suggestion
Qualification badges are another route for RSEs to find people with the same technical interests.
```
Instead of "route" this could also, perhaps, say "option"?

Reviewer 2: yeah, why not? I think we haven't used option so far in the paper....

Reviewer 2: included option.

This paper started from a community workshop at deRSE23 in Paderborn
where people working in RSE related fields got together to figure out
structures and ideas for educating newcomers to this field.
One outcome of this diverse gathering is that RSEs from far away fields gather
around similar core concepts but at the same time share a vision of how to
update the scientific ecosystem to the age of digitalization.
In this publication, we have tried to formalise these concepts.
We have formulated a set of values that guide our actions in society

> to manifestly make RSEs part of the scientific community but at the same time,

Reviewer 2: @jlinx try to find a good equivalent for wissenschaftlicher Wertegemeinschaft....

Reviewer: This could just say "...the wider research community..." rather than scientific community.

I see that Google translates "wissenschaftlicher Wertegemeinschaft" as scientific community of values but I guess this ultimately comes back to our discussion about "wissenschaft" and the difference between science and research?

Reviewer 2: What in my opinion gets lost is the notion of being part of a group of people that share the same values. Scientific community in contrast has no *explicit* notion of  shared values.

Reviewer 2: part of the scientific community sharing the common values.

> being Software Engineers, we cherish that we have to take responsibility for our tools.

Reviewer: I just spotted this while looking a the changes - I'm not sure that "cherish" is the right word here - I might just say "recognise"
```suggestion
being Software Engineers, we recognise that we have to take responsibility for our tools.
```

Reviewer 2: I guess we had meaning 2 over here in mind: https://www.oxfordlearnersdictionaries.com/definition/english/cherish
